pkg/webhook/gitmanager/init.go

```diff
@@ -39,14 +50,263 @@ type CustomRepo struct {
 	return &cr, nil
 }
+
+func (cr *CustomRepo) createRepo(storer *memory.Storage) (*git.Repository, error) {
+	if cr.StorageMode == StorageModeInMemory {
+		r, err := git.Init(storer, cr.Fs)
+		if err == git.ErrRepositoryAlreadyExists {
+			klog.V(2).InfoS("network policy repository already exists - skipping initialization")
+			return nil, err
+		} else if err != nil {
+			klog.ErrorS(err, "unable to initialize git repo")
+			return nil, err
+		}
+		return r, nil
+	}
+	if cr.Dir == "" {
+		path, err := os.Getwd()
+		if err != nil {
+			klog.ErrorS(err, "unable to retrieve the current working directory")
+			return nil, err
+		}
+		if path != "/" {
+			cr.Dir = path
+		}
+	}
+	cr.Dir += "/resource-auditing-repo"
+	r, err := git.PlainInit(cr.Dir, false)
+	if err == git.ErrRepositoryAlreadyExists {
+		klog.V(2).InfoS("network policy repository already exists - skipping initialization")
+		r, err := git.PlainOpen(cr.Dir)
+		if err != nil {
+			klog.ErrorS(err, "unable to retrieve existing repository")
+			return nil, err
+		}
+		return r, git.ErrRepositoryAlreadyExists
+	} else if err != nil {
+		klog.ErrorS(err, "unable to initialize git repo")
+		return nil, err
+	}
+	return r, nil
+}
+
+func (cr *CustomRepo) addResources() error {
+	if cr.StorageMode == StorageModeDisk {
+		os.Mkdir(cr.Dir+"/k8s-policies", 0700)
+		os.Mkdir(cr.Dir+"/antrea-policies", 0700)
+		os.Mkdir(cr.Dir+"/antrea-cluster-policies", 0700)
+		os.Mkdir(cr.Dir+"/antrea-tiers", 0700)
+	} else {
+		cr.Fs.MkdirAll("k8s-policies", 0700)
+		cr.Fs.MkdirAll("antrea-policies", 0700)
+		cr.Fs.MkdirAll("antrea-cluster-policies", 0700)
+		cr.Fs.MkdirAll("antrea-tiers", 0700)
+	}
+	if err := cr.addK8sPolicies(); err != nil {
+		klog.ErrorS(err, "unable to add K8s network policies to repository")
+		return err
+	}
+	if err := cr.addAntreaPolicies(); err != nil {
+		klog.ErrorS(err, "unable to add Antrea network policies to repository")
+		return err
+	}
+	if err := cr.addAntreaClusterPolicies(); err != nil {
+		klog.ErrorS(err, "unable to add Antrea cluster network policies to repository")
+		return err
+	}
+	if err := cr.addAntreaTiers(); err != nil {
+		klog.ErrorS(err, "unable to add Antrea tiers to repository")
+		return err
+	}
+	return nil
+}
+
+func (cr *CustomRepo) addK8sPolicies() error {
+	list := &unstructured.UnstructuredList{}
+	list.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "networking.k8s.io",
+		Version: "v1",
+		Kind:    "NetworkPolicyList",
+	})
+	policies, err := cr.K8s.ListResource(list)
+	if err != nil {
+		return err
+	}
+	var namespaces []string
+	for i, np := range policies.Items {
+		clearFields(&policies.Items[i])
+		name := np.GetName()
+		namespace := np.GetNamespace()
+		if !stringInSlice(namespace, namespaces) {
+			namespaces = append(namespaces, namespace)
+			if cr.StorageMode == StorageModeDisk {
+				os.Mkdir(cr.Dir+"/k8s-policies/"+namespace, 0700)
+			} else {
+				cr.Fs.MkdirAll("k8s-policies/"+namespace, 0700)
+			}
```

Comment:
this is handled by `addResources`, why do we need this?

Reply:
addResources handles adding the resource directories (/k8s-policies, /antrea-policies, /antrea-cluster-policies, and /antrea-tiers), but this logic is meant to handle adding new namespaces as subdirectories of these resource directories. Looking back at it now, maybe it would be cleaner to just query the go-client for namespaces and then add those subdirectories to all base directories, regardless of whether a resource exists for that particular resource directory/namespace directory (but then there might be many unnecessary namespace folders with no resources in them)

Reply:
I think the current solution is fine, I read the code too fast